### PR TITLE
MINOR: Add links to connector configs in TOC

### PR DESCRIPTION
**Pull request by mimaison**
Add links to Sink and Source connectors configurations

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: docs/toc.html [hunk 1]
```diff
@@ -45,6 +45,10 @@
                 <li><a href="#producerconfigs">3.3 Producer Configs</a>
                 <li><a href="#consumerconfigs">3.4 Consumer Configs</a>
                 <li><a href="#connectconfigs">3.5 Kafka Connect Configs</a>
+                    <ul>
+                        <li><a href="#sourceconnectconfigs">Source Connector Configs</a>
+                        <li><a href="#sinkconnectconfigs">Sink Connector Configs</a>
+                    </ul>
```
Comment (showuon) [inline docs/toc.html:51]: It's good to have these 2 sub categories in TOC. But usually with this TOC, we will think there are only 2 categories under `3.5 Kafka Connect Configs`. However, there is another sub-category `Kafka Connect Configs` there. Do you think we can make the TOC as below:
- 3.5 Kafka Connect Configs
   - Kafka Connect Configs
   - Source Connector Configs
   - Sink Connector Configs

And maybe change `3.5.1` and `3.5.2` in front of `Source Connector Configs` and `Sink Connector Configs`. WDYT?

**Review by showuon — COMMENTED** [review]
Thanks for the PR. Left a comment. Thanks.

File: docs/toc.html (hunk 1 above)
Comment (mimaison) [reply docs/toc.html:51]: I considered that but then both `Kafka Connect Configs` links would point to the same anchor. Also `Kafka Connect Configs` is one level higher (3.5) level than the other two sections (3.5.1 and 3.5.2).

About adding `3.5.1` and `3.5.2`, we don't include numbers for sub categories, for example see `[Adding and removing topics](https://kafka.apache.org/documentation/#basic_ops_add_topic)` in the TOC, so I've kept it consistent with the other ones.

**Review by andrewchoi5 — APPROVED** [review]

File: docs/toc.html (hunk 1 above)
Comment (showuon) [reply docs/toc.html:51]: OK, I checked the other sub category links, like: 
 - https://kafka.apache.org/documentation/#os 
 - https://kafka.apache.org/documentation/#basic_ops_add_topic
They all jump to the sub category directly, while there are still some paragraph above it. So, I agree with you. We can add sub TOC: `Source Connector Configs` and `Sink Connector Configs` into `Kafka Connect Configs`, and there keep the rest as is. 
Thanks.

**Review by showuon — APPROVED** [review]
LGTM!

**Pull request merged**